sql/moz-fx-data-shared-prod/mozilla_vpn_derived/acquisition_funnel_v1/query.sql

```diff
@@ -0,0 +1,107 @@
+WITH logged_in AS (
+  SELECT
+    fxa_uid,
+    MIN(created_at) AS first_logged_in,
+  FROM
+    mozilla_vpn_external.users_v1
+  GROUP BY
+    fxa_uid
+),
+viewed_login AS (
+  SELECT
+    fxa_uid,
+    MIN(flow_started) AS first_viewed_login,
+  FROM
+    login_flows_v1
+  FULL JOIN
+    logged_in
+  USING
+    (fxa_uid)
+  WHERE
+    -- include entries in users_v1 that are not present in login_flows_v1
+    first_logged_in IS NULL
+    -- TODO: exclude login on desktop client, and second login attempt on mobile.
+    -- As an approximation, filter out flows that started after first login.
+    OR first_logged_in > flow_started
+  GROUP BY
+    fxa_uid,
+    -- when fxa_uid is missing, produce one row per flow_id
+    COALESCE(fxa_uid, flow_id)
+),
+subscribed AS (
+  SELECT
+    mozfun.map.get_key(metadata, "userid") AS fxa_uid,
+    MIN(subscriptions.event_timestamp) AS first_subscribed,
+  FROM
+    stripe.subscriptions
+  JOIN
+    stripe.plans
+  ON
+    (subscriptions.plan = plans.id)
+  JOIN
+    stripe.products
+  ON
+    (plans.product = products.id)
+  JOIN
+    stripe_external.customers_v1
+  ON
+    (subscriptions.customer = customers_v1.id)
+  WHERE
+    products.name = "Mozilla VPN"
+    AND subscriptions.status NOT IN ("incomplete", "incomplete_expired")
+  GROUP BY
+    fxa_uid
+),
+registered_device AS (
+  SELECT
+    fxa_uid,
+    MIN(devices_v1.created_at) AS first_registered_device,
+  FROM
+    -- TODO: validate assumption that devices are not deleted unless the user is deleted
+    mozilla_vpn_external.devices_v1
+  JOIN
+    mozilla_vpn_external.users_v1
+  ON
+    (users_v1.id = devices_v1.user_id)
+  GROUP BY
+    fxa_uid
+),
+activated_vpn AS (
+  -- TODO: this is undercounted, and may not be fully measurable
+  SELECT
+    fxa_uid,
+    first_activated_vpn,
+  FROM
+    activated_vpn_v1
```

Comment:
You didn't `GROUP BY fxa_uid` here - intentional because you know it's unique in `activated_vpn_1`, or an oversight?

Reply:
because I know it's unique